src/box/lua/net_box.lua

```diff
@@ -766,7 +769,16 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
-    return (pcall(self._request, self, M_PING, opts, nil, self._stream_id))
+    local status, ret_val = pcall(self._request, self, M_PING, opts, nil, self._stream_id)
+    if not status and ret_val then
+        if type(ret_val) == 'cdata' and (ret_val:unpack().code == E_NO_CONNECTION or
+                                         ret_val:unpack().type == 'TimedOut') then
```

Comment:
Lets not do the unpack 2 times. You don't even need the unpack really. Error object can return `.code` and `.type` keys directly.